s4e-backend/src/main/java/pl/cyfronet/s4e/service/SearchService.java

```diff
@@ -43,6 +58,32 @@ public Long getCountBy(Map<String, Object> params) throws SQLException, QueryExc
         return repository.countAllByParamsMap(params);
     }
 
+    public void checkProductTypePresent(Map<String, Object> params) throws BadRequestException {
+        if (params.get(PRODUCT_TYPE) == null) {
+            throw new BadRequestException("No product type.");
+        }
+    }
+
+    public void applyLicenseByProductType(Map<String, Object> params) throws NotFoundException {
```

Comment:
Upon entry to this method I would strip any existing accessType key present in params, just to be safe it's value wasn't injected.